src/components/mapexplorer.js

```diff
@@ -80,9 +80,16 @@ function MapExplorer({
         const region = getRegionFromState(
           states.find((state) => name === state.state)
         );
-        setCurrentHoveredRegion(region);
-        setPanelRegion(region);
-        onMapHighlightChange(region);
+        if (region) {
+          setCurrentHoveredRegion(region);
```

Comment:
avoid duplicate function calls. **setCurrentHoveredRegion**, **setPanelRegion**, **onMapHighlightChange** are being referred twice with just a change in param region and noregion.
Write them once and manipulate the value passed to them as.
`
let region ;
if (!region){
   region = noregion;
}
setCurrentHoveredRegion(region);
setPanelRegion(region);
onMapHighlightChange(region)
`